c++/src/capnp/compiler/capnpc-c++.c++

```diff
@@ -3045,7 +3054,12 @@ private:
           "#if CAPNP_VERSION != ", CAPNP_VERSION, "\n"
           "#error \"Version mismatch between generated code and library headers.  You must "
               "use the same version of the Cap'n Proto compiler and library.\"\n"
-          "#endif\n"
+          "#endif\n",
+          omitSchemas ? kj::strTree(
+                  "#if !CAPNP_LITE\n",
+                  "#error \"$Cxx.omitSchemas requires that you compile using CAPNP_LITE\"\n"
```

Comment:
Why does it require this? I would expect it to work fine without CAPNP_LITE (as long as you don't try to use the schema or dynamic APIs with the type, and all dependent types also omit schemas). Is there a particular error you're seeing?

Reply:
You are correct, I added this before I fully understood the generated code, After running some tests it seems it is not required.